AzureCalling/Controllers/CallViewController.swift

```diff
@@ -462,37 +499,28 @@ class CallViewController: UIViewController, UICollectionViewDelegate, UICollecti
         var indexPathMoves: [(at: IndexPath, to: IndexPath)] = []
         var insertIndexPaths: [IndexPath] = []
 
-        // Build new maps and collect changes
-        for (index, participant) in callingContext.displayedRemoteParticipants.enumerated() {
-            let userIdentifier = participant.identifier.stringValue ?? ""
-            let indexPath = IndexPath(item: index, section: 0)
-            var participantView: ParticipantView
-
-            // Check for previously tracked participants
-            if let prevIndexPath = prevParticipantIdIndexPathMap[userIdentifier],
-               let prevParticipantView = prevParticipantIndexPathViewMap[prevIndexPath] {
-                prevParticipantIndexPathViewMap.removeValue(forKey: prevIndexPath)
-
-                participantView = prevParticipantView
-
-                if prevIndexPath != indexPath {
-                    // Add to move list
-                    indexPathMoves.append((at: prevIndexPath, to: indexPath))
-                }
-            } else {
-                participantView = ParticipantView()
-
-                // Add to insert list
-                insertIndexPaths.append(indexPath)
+        // Check to see if any screenshare
+        for participant in callingContext.displayedRemoteParticipants {
```

Comment:
Is that possible replace the callingContext.displayedRemoteParticiopants with a local cache like callingViewController.wantToDisplayedRemoteParticipant. When there is a screensharing user we just put the user into the local cache, otherwise, the localCache would contain the callContext,displayedRemoteParticipants.

And when we for loop, we for loop the local cache callingViewController.wantToDisplayedRemoteParticipant to update the participant view

Reply:
Please see new commit that has new logic in calling Context onVideoStreamsUpdated

Reply:
The logic change of filtering out the screensharing user doesn't affect the approach I suggest. 
In the approach I mentioned above. You don't need to abstract any helper class and cache the prevIndexPath map, you only need to update the iterate array on the entry of updateParticipantView

Reply:
Revert to the previous implementation to not have the new class variables in the new commit